content/blog/2023-useful-documentation-adrs/index.mdx

```diff
@@ -0,0 +1,74 @@
+
+* Contracts between services. Network protocols, data formats, etc. You'd document it if that was _your team's_ decision.
+
+* Anything that you've created a _spike_ or _proof of concept_ for. If this works rests only in Jira, I have a feeling it might get lost. Might be a good idea to co-locate it alongside your code. Whether the result was _not_ to do something, do it, or wait.
+
+
+## Why would I want to document that?
+
+It will help you make decisions. An ADR needs to contain the _context_ around the decision. So, you'll need to clear up exactly what problem you're trying to solve (decide how to solve) and specify what options you've considering. Once you write that down, it's usually much easier to make (propose) the decision. Probably because writing _is_ thinking.
+
+Once somebody worked on the problem, did the research and has a proposal, you can forward the in-progress document to your team as the agenda for a group discussion. This will let everybody come prepared and know what you did (and didn't) consider when proposing a decision.
+
+It stops you from making blindly making and changing decisions. Writing down what options where considered helps consider all the alternatives and writing down your reasoning prevents you from overturning this decision in the future on a whim.
+
+Finally, it saves onboarding time. Often, when a new member joins a team they might wonder why something is built a particular way, or try to port over their favourite solutions from previous projects. Having clear ADRs that explain your patterns and choices helps new hires quickly learn the reasoning behind them and see whether there's a better solution that they could offer.
+
+
+## I'm sold, how would I want to document that?
+
+First, setup a place for your ADRs, probably somewhere in the `docs` folder in your repo. You have a `docs` folder, right?
+
+Second, ADRs are time-bound, meaning you've made a decision at a particular point in time and it's important to track that. Of course, you can always check the date in the document, but I find it helpful to number the filenames. For example: `1-i18n-library.md`, `2-state-management-provider.md`, etc.
+
+Third, decide on a format. You can [pick one of the popular ones](https://github.com/joelparkerhenderson/architecture-decision-record/tree/main/templates) or write your own. It doesn't matter that much which one you pick, but having _a_ structure will make it easier to write them.
```

Comment:
Link broken, did you mean https://github.com/joelparkerhenderson/architecture-decision-record/tree/main/locales/en/templates ?

Reply:
They move the files around overnight 🤦‍♂️
Thanks!